### PR TITLE
gPTP packet parsing fixes

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

**Review by tbursztyka — APPROVED** [review]

**Comment by jukkar** [comment]
Same issue in sam-e70 so I will fix that too in this PR.
Also I started to see
```
[00:00:10.621,648] <wrn> net_gptp.gptp_md_follow_up_receipt_timeout: No FOLLOWUP received after SYNC message
```

which requires some more investigation.

File: drivers/ethernet/eth_mcux.c [hunk 1]
```diff
@@ -394,19 +396,37 @@ static bool eth_get_ptp_data(struct net_if *iface, struct net_pkt *pkt,
...
 		/* Cannot use GPTP_HDR as net_pkt fields are not all filled */
-		struct gptp_hdr *hdr = gptp_get_hdr(pkt);
+		struct gptp_hdr *hdr;
+
+		/* In TX, the first net_buf contains the Ethernet header
+		 * and the actual gPTP header is in the second net_buf.
+		 * In RX, the Ethernet header + other headers are in the
+		 * first net_buf.
+		 */
+		if (is_tx == true) {
```
Comment (pfalcon) [inline drivers/ethernet/eth_mcux.c]: Well, that's too much IMHO ;-). Boolean values should be tested as `(booly)` and `(!booly)`.

File: drivers/ethernet/eth_mcux.c [hunk 2]
```diff
@@ -394,19 +396,37 @@ static bool eth_get_ptp_data(struct net_if *iface, struct net_pkt *pkt,
...
 	}
 
 	net_pkt_set_priority(pkt, NET_PRIORITY_CA);
 
 	if (ptpTsData) {
 		/* Cannot use GPTP_HDR as net_pkt fields are not all filled */
-		struct gptp_hdr *hdr = gptp_get_hdr(pkt);
+		struct gptp_hdr *hdr;
+
+		/* In TX, the first net_buf contains the Ethernet header
```
Comment (pfalcon) [inline drivers/ethernet/eth_mcux.c]: Well, I see that @tbursztyka approved this, but any comments? This looks too adhoc/fragile.

File: drivers/ethernet/eth_mcux.c (hunk 1 above)
Comment (jukkar) [reply drivers/ethernet/eth_mcux.c]: I agree, but wasn't it some misra rule that says we need to do full check here?

**Comment by jukkar** [comment]
Updated sam-e70 and native_posix eth drivers.
Also the warning I saw depends on the system setup thus it is nominal in this case.

File: drivers/ethernet/eth_mcux.c (hunk 1 above)
Comment (pfalcon) [reply drivers/ethernet/eth_mcux.c]: To the best of my knowledge, MISRA requires that boolean expressions are used in if, while, etc. conditions. A boolean variable itself is a boolean expression. Calling to @ceolin for clarification.

Comment (pfalcon) [reply drivers/ethernet/eth_mcux.c]: (Otherwise, I grepped the codebase, and we have that pattern crept in already...)

Comment (ceolin) [reply drivers/ethernet/eth_mcux.c]: @pfalcon you're right, a boolean variable is a boolean expression. No need explicit comparison.

**Comment by jukkar** [comment]
Updated according to comments

**Pull request merged**